lib/AST/DeclContext.cpp

```diff
@@ -1447,16 +1448,28 @@ bool AccessScope::allowsPrivateAccess(const DeclContext *useDC, const DeclContex
       return usePkg->isSamePackageAs(srcPkg);
     }
   }
-  // Do not allow access if the sourceDC is in a different file
-  auto useSF = useDC->getOutermostParentSourceFile();
-  if (useSF != sourceDC->getOutermostParentSourceFile())
-    return false;
 
   // Do not allow access if the sourceDC does not represent a type.
   auto sourceNTD = sourceDC->getSelfNominalTypeDecl();
   if (!sourceNTD)
     return false;
 
+  // Do not allow access if the sourceDC is in a different file
+  auto useSF = useDC->getOutermostParentSourceFile();
+  if (useSF != sourceDC->getOutermostParentSourceFile()) {
+    // This might be a C++ declaration with a SWIFT_PRIVATE_FILEID
+    // attribute, which asks us to treat it as if it were defined in the file
+    // with the specified FileID.
+
+    auto clangDecl = sourceNTD->getDecl()->getClangDecl();
+    if (!clangDecl)
+      return false;
+
+    auto blessedFileID = importer::getPrivateFileIDAttrs(clangDecl);
+    if (blessedFileID.empty() || !useSF->matchesFileID(blessedFileID[0].first))
```

Comment:
The importer is supporting multiple attributes, so we should loop over all of them here.